node/forest_libp2p/src/service.rs

```diff
@@ -187,13 +187,15 @@ where
                                 response,
                             }).await;
                         }
-                        ForestBehaviourEvent::BlockSyncRequest { channel, .. } => {
-                            // TODO implement blocksync provider
-                            let _ = channel.send(BlockSyncResponse {
-                                chain: vec![],
-                                status: 203,
-                                message: "handling requests not implemented".to_owned(),
-                            }).await;
+                        ForestBehaviourEvent::BlockSyncRequest { channel, peer, request } => {
+                            debug!("Received blocksync request (peerId: {:?})", peer);
+                            let db = self.db.clone();
+                            async {
+                                let response = task::spawn_blocking(move || -> BlockSyncResponse {
+                                    make_blocksync_response(db, &request.clone())
```

Comment:
```suggestion
                                    make_blocksync_response(db, &request)
```
This request can be moved, no need to clone

Reply:
updated.